sklearn/svm/_base.py

```diff
@@ -671,20 +674,11 @@ def _get_coef(self):
         return safe_sparse_dot(self._dual_coef_, self.support_vectors_)
 
     @property
-    def n_support_(self):
```

Comment:
Looking at this again, I think it makes sense to have `n_support_` for regressors and not deprecate it.

WDYT @micky774?

Reply:
I'm not really a fan of it in the context of `(Nu)SVR` as-is. Its usage in other `SVM` estimators as a `(num_classes, )` tuple seems to imply that the regressors operate with a single class which imo is not quite fair. 

With that being said I would be more friendly to it if we also alter the documentation from
> Number of support vectors for each class.

to
> Number of support vectors.

just to further clarify the distinct semantics regarding the shape of the attribute in regressors.

Reply:
Now that are we not deprecating `n_support_` anymore, we technically do not need to do the `n_support_` refactor. To make this a quicker review for others, we can revert the changes for `n_support_` and keep the `class_weight_` deprecation.

Let's open a small second PR to update `n_support_` documentation.

Reply:
Opened here: #23353